### DISCUSSION TITLE
2 Slingboxes - Raspberry Pi - Server Startup

### DISCUSSION THREAD

**Discussion by socelillyhammer** [GerryDazoo/Slinger#474] [Q&A]
Found a small issue with Raspberry Pi running 2 slingboxes on 1 config file.

following the readme and enabling the sling.service and starting the sling.service on a reboot the server seems to encounter an error finding the slingboxes.

To fix this issue I have to login to the remote and restart the server by pushing the restart button.  That seems to fix the issue.

While only running 1 slingbox on the server, on the reboot of the RPi i did not encounter this issue.

This is the data from my logfile showing that after I kill the server, the server finds the two boxes

Any thoughts would be appreciated.  Thanks in advanced

BOXES [('sb1', 'basement'), ('sb2', 'master')]
Streamer Running:  Streamer Running: 10  10config.ini  config.inibasement  masterbasement  master51820  518208388608 
8388608
Checking for slingbox atChecking for slingbox at  ('192.168.1.189', 5206)('192.168.1.182', 5208)

**### starting up on  port 51820Error connecting to Error connecting to  ('192.168.1.189', 5206) 
('192.168.1.182', 5208)**
 8388608
**master basementNo valid slingbox ip info found in config.ini 
No valid slingbox ip info found in config.ini
Can't find a slingbox on network. Please make sure it's plugged in and connected. Check config.iniCan't find a slingbox on network. Please make sure it's plugged in and connected. Check config.ini**

Building page for basement
Reading Custom Remote definition from remote.txt
Error reading remote definition file, using defaults
Building page for master
Reading Custom Remote definition from remote.txt
Error reading remote definition file, using defaults
 * Serving Flask app "slingbox_server" (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
basement Giving up. Sorry...
master Giving up. Sorry...
01/03/2023, 00:42:00.212   RemoteControl connection from ('192.168.1.60', 55365)

Starting remote control stream handler for  ('192.168.1.60', 55365) to port 51821
Remote Control Connected
GET Remote/basement
Exiting Remote Control Stream Handler for ('192.168.1.60', 55365)
01/03/2023, 00:42:01.625   RemoteControl connection from ('192.168.1.60', 55367)

Starting remote control stream handler for  ('192.168.1.60', 55367) to port 51821
Remote Control Connected
Key Data ('Restart', 'Restart')
Key Data ('Digits', '')
**Restarting, killing myself**

Version : 3.08f Running on Linux-5.15.76-v7l+-armv7l-with-glibc2.31 pid= 1786 slingbox_server.py
Using config file config.ini
Connection Manager Running on port 51820 with 10 max streams using URL slingbox.
BOXES [('sb1', 'basement'), ('sb2', 'master')]
Streamer Running: Streamer Running:  starting up on  port 51820 10 config.ini10   master config.ini8388608master
  51820 8388608
basement basement 51820Checking for slingbox at  8388608('192.168.1.189', 5206)

Checking for slingbox at ('192.168.1.182', 5208)
('192.168.1.182', 5208) OK
basement Using slingbox at  ('192.168.1.182', 5208)
Streamer:  basement Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
('192.168.1.189', 5206) OK
master Using slingbox at  ('192.168.1.189', 5206)
Streamer:  master Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
Building page for basement
Reading Custom Remote definition from remote.txt
Error reading remote definition file, using defaults
Building page for master
Reading Custom Remote definition from remote.txt
Error reading remote definition file, using defaults
 * Serving Flask app "slingbox_server" (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off

**Comment by GerryDazoo** [comment]
The problem is your network is slow in coming up and when slinger starts it can't connect.

Try the latest service file from the repository. If that doesn't work you'll need to wait for the next release where you can add a ConnectRetries=1  into your slingbox definitions section or stick a small delay into the service exe definition before starting the server.
